src/dataflow/src/server.rs

```diff
@@ -297,41 +297,104 @@ pub struct Config {
     pub metrics_registry: MetricsRegistry,
     /// Handle to the persistence runtime. None if disabled.
     pub persist: Option<RuntimeClient>,
-    /// Responses to commands should be sent into this channel.
-    pub feedback_tx: mpsc::UnboundedSender<Response>,
+    /// An optional callback that is presented with both halves of the dataflow
+    /// feedback channel on startup. The callback can swap the channel for
+    /// another in order to intercept messages.
+    ///
+    /// For testing purposes only.
+    pub response_interceptor: Option<ResponseInterceptor>,
```

Comment:
Nit, but this feels a bit like a leaky abstraction that we'll want to fix. I prefer it more than the duplicated coord code, but .. I suspect we'll want to figure out how to shoo this out of the abstraction here eventually. If it is expected to be more durable than that, we should probably chat about what the expectations for it are!

Reply:
Yeah, 100%! I figure we'll want `trait DataflowClient` in coord at some point soon, and then we can pass in an `InterceptingDataflowClient` or whatever else `coordtest` wants.